src/libraries/System.Net.Http/src/System/Net/Http/Headers/ViaHeaderValue.cs

```diff
@@ -169,11 +169,10 @@ internal static int GetViaLength(string? input, int startIndex, out object? pars
 
             // If we reached the end of the string after reading protocolName/Version we return (we expect at least
             // <receivedBy> to follow). If reading protocolName/Version read 0 bytes, we return.
-            if ((current == startIndex) || (current == input.Length))
+            if ((current == startIndex) || (current == input.Length) || (protocolVersion == null))
```

Comment:
```suggestion
            if ((current == 0) || (current == input.Length))
```

I think checking for 0 and keeping the assert is what this was supposed to look like